### PR TITLE
Replace Actions API handlers in entities with RTK Query

### LINKED ISSUE #50873: Replace Actions API handlers in entities with RTK Query
- Make the entity framework use RTK query under the hood for all Actions API requests
- Port Actions to use new `EntityObjectLoaderRtkQuery` (#50868)

----

_We missed this before, it should have been done in Milestone 1 in #40788._

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #50873

### Description

We missed this before, it should have been done in Milestone 1 in https://github.com/metabase/metabase/issues/40788.

This PR also enables the new `EntityObjectLoaderRtkQuery` for Actions (that's why it's not going to be backported).

### How to verify

CI is green.

### CODE REVIEW

File: frontend/src/metabase/api/action.ts [hunk 1]
```diff
@@ -1,21 +1,73 @@
...
+      providesTags: action => (action ? provideActionTags(action) : []),
+    }),
+    createAction: builder.mutation<WritebackAction, CreateActionRequest>({
+      query: body => ({
+        method: "POST",
+        url: "/api/action",
+        body,
+      }),
+      invalidatesTags: (action, error) =>
+        action ? [...invalidateTags(error, [listTag("action")])] : [],
```
Comment (romeovs) [inline frontend/src/metabase/api/action.ts:46]: Can we just write?

```suggestion
        action ? invalidateTags(error, [listTag("action")]) : [],
```

File: frontend/src/metabase/api/action.ts [hunk 2]
```diff
@@ -1,21 +1,73 @@
...
+      }),
+      invalidatesTags: (action, error) =>
+        action
+          ? [
+              ...invalidateTags(error, [
+                listTag("action"),
+                idTag("action", action.id),
+              ]),
+            ]
+          : [],
```
Comment (romeovs) [inline frontend/src/metabase/api/action.ts:62]: Can we simplify to
```suggestion
        action
          ? invalidateTags(error, [
              listTag("action"),
              idTag("action", action.id),
            ])
          : [],
```

File: frontend/src/metabase-types/api/actions.ts [hunk 3]
```diff
@@ -5,6 +5,47 @@ import type { Parameter, ParameterId, ParameterTarget } from "./parameters";
...
+  model_id?: CardId;
+  name?: string;
+  parameter_mappings?: Record<ParameterId, ParameterTarget>;
+  parameters?: WritebackParameter[];
+  response_handle?: string | null;
+  template?: HttpActionTemplate;
+  url?: string;
+  body?: string;
+  headers?: string;
+  type?: "query" | "implicit" | "http";
```
Comment (romeovs) [inline frontend/src/metabase-types/api/actions.ts:45]: Could this be a separate type that's reused in both `CreateActionRequest` and `UpdateActionRequest`?

**Review by romeovs — APPROVED** [review]
Some nits, but otherwise looks great!

File: frontend/src/metabase/api/action.ts (hunk 1 above)
Comment (kamilmielnik) [reply frontend/src/metabase/api/action.ts:46]: 6568d593090107429e3196443f19b935c3da9c23

File: frontend/src/metabase/api/action.ts (hunk 2 above)
Comment (kamilmielnik) [reply frontend/src/metabase/api/action.ts]: 6568d593090107429e3196443f19b935c3da9c23

File: frontend/src/metabase-types/api/actions.ts (hunk 3 above)
Comment (kamilmielnik) [reply frontend/src/metabase-types/api/actions.ts]: Actually we should remove `type` from `UpdateActionRequest` as [it's not supported](https://github.com/metabase/metabase/blob/6568d593090107429e3196443f19b935c3da9c23/frontend/src/metabase/api/action.ts#L52).

Done: 7130104e7b573b9defbf7610ea25d4ab74fd7d6a

**Pull request merged**